### ISSUE TITLE
Cascade layers with nesting

### ISSUE THREAD

**Issue by mayank99** [parcel-bundler/lightningcss#374]
The CSS nesting spec is getting finalized soon, and it supports nesting all at-rules, including `@layer`:

```css
.foo {
  @layer A {
    background: red;
  }
}
/* equivalent to: @layer A { .foo { background: red; } } */
```

Currently this does not work in lightningcss (see [playground](https://lightningcss.dev/playground/index.html#%7B%22minify%22%3Afalse%2C%22nesting%22%3Atrue%2C%22customMedia%22%3Afalse%2C%22cssModules%22%3Afalse%2C%22analyzeDependencies%22%3Afalse%2C%22targets%22%3A%7B%22chrome%22%3A6225920%7D%2C%22source%22%3A%22.foo%20%7B%5Cn%20%20%40media%20(prefers-color-scheme%3A%20dark)%20%7B%5Cn%20%20%20%20background%3A%20rebeccapurple%3B%5Cn%20%20%7D%5Cn%5Cn%20%20%40layer%20A%20%7B%5Cn%20%20%20%20background%3A%20hotpink%3B%5Cn%20%20%7D%5Cn%5Cn%20%20%40layer%20B%20%7B%5Cn%20%20%20%20%40media%20(prefers-color-scheme%3A%20dark)%20%7B%5Cn%20%20%20%20%20%20background%3A%20darkgray%3B%5Cn%20%20%20%20%7D%5Cn%20%20%7D%5Cn%7D%5Cn%22%2C%22unusedSymbols%22%3A%5B%5D%2C%22version%22%3A%22local%22%7D)).

I currently use [`postcss-nested`](https://github.com/postcss/postcss-nested) for this but would like to switch to lightningcss.

**Comment by devongovett** [comment]
I think this was fixed by 615893c00386fe154c7ad33fddc116416e25b407 but it hasn't been released yet. I'm planning a release for next week so it should go out with that.

**Comment by mayank99** [comment]
awesome, then I'll close this and wait for the release. will reopen if it doesn't work

**Comment by devongovett** [comment]
Released in v1.18.0.

**Comment by mayank99** [comment]
works great, thanks!
